File: java/src/main/java/com/myriad/wdl/model/sections/WdlOutput.java
package com.myriad.wdl.model.sections;

import com.myriad.wdl.model.definitions.WdlTask.WdlTaskElement;
import com.myriad.wdl.model.definitions.WdlWorkflow.WdlWorkflowElement;
import com.myriad.wdl.model.statements.WdlDeclaration.WdlBoundDeclaration;
import java.util.ArrayDeque;

/**
 * Output section node.
 *
 * <p>This models the explicit `output { ... }` section that may appear in tasks and workflows.
 * Output declarations are always bound to expressions.
 */
public final class WdlOutput implements WdlTaskElement, WdlWorkflowElement {
  private final ArrayDeque<WdlBoundDeclaration> elements = new ArrayDeque<>();

  /** Returns the ordered bound declarations contained in the output section. */
  public ArrayDeque<WdlBoundDeclaration> elements() {
    return elements;
  }
}

File: typescript/src/model/sections/wdl-output.ts
/** Output section nodes. */
import type { WdlTaskElement } from '../definitions/wdl-task.js';
import type { WdlWorkflowElement } from '../definitions/wdl-workflow.js';
import type { WdlBoundDeclaration } from '../statements/wdl-declaration.js';

/** Models an explicit `output { ... }` section in a task or workflow. */
export class WdlOutput implements WdlTaskElement, WdlWorkflowElement {
  private readonly elementValues: WdlBoundDeclaration[] = [];
  /** Returns the ordered bound declarations contained in the output section. */
  public elements(): WdlBoundDeclaration[] {
    return this.elementValues;
  }
}
